app/models/setting/email.rb

```diff
@@ -0,0 +1,72 @@
+require 'facter'
+
+class Setting::Email < Setting
+  WHITELIST = %w(send_welcome_email email_reply_address email_subject_prefix)
+
+  def self.load_defaults
+    # Check the table exists
+    return unless super
+    domain = Facter.value(:domain) || SETTINGS[:domain]
+    email_reply_address = "Foreman-noreply@#{domain}"
+
+    self.transaction do
+      [
+        self.set('email_reply_address', N_("Email reply address for emails that Foreman is sending"), email_reply_address, N_('Email reply address')),
+        self.set('email_subject_prefix', N_("Prefix to add to all outgoing email"), '[foreman]', N_('Email subject prefix')),
+        self.set('send_welcome_email', N_("Send a welcome email including username and URL to new users"), false, N_('Send welcome email')),
+        self.set('delivery_method', N_("Method used to deliver e-mail"), 'sendmail', N_('Delivery method'), nil, { :collection => Proc.new {{'sendmail' => :sendmail, 'smtp' => :smtp}}}),
+        self.set('smtp_enable_starttls_auto', N_("Use STARTTLS automatically"), false, N_('SMTP enable StartTLS auto')),
+        self.set('smtp_tls', N_("Enable SSL/TLS"), false, N_('SMTP use SSL/TLS')),
+        self.set('smtp_address', N_("Address to connect to"), '', N_('SMTP address')),
+        self.set('smtp_port', N_("Port to connect to"), 25, N_('SMTP port')),
+        self.set('smtp_domain', N_("Email domain"), '', N_('SMTP email domain')),
```

Comment:
The description is ambiguous, this is the EHLO domain, right? Not the from domain?